cmd/openshift-install/create.go

```diff
@@ -278,7 +288,50 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
 	return destroybootstrap.Destroy(rootOpts.dir)
 }
 
-// waitForconsole returns the console URL from the route 'console' in namespace openshift-console
+// waitForInitializedCluster watches the ClusterVersion waiting for confirmation
+// that the cluster has been initialized.
+func waitForInitializedCluster(ctx context.Context, config *rest.Config) error {
+	timeout := 30 * time.Minute
+	logrus.Infof("Waiting up to %v for the cluster to initialize...", timeout)
+	cc, err := configclient.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "failed to create a config client")
+	}
+	clusterVersionContext, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	_, err = clientwatch.UntilWithSync(
+		clusterVersionContext,
+		cache.NewListWatchFromClient(cc.ConfigV1().RESTClient(), "clusterversions", "", fields.OneTermEqualSelector("metadata.name", "version")),
+		&configv1.ClusterVersion{},
+		nil,
+		func(event watch.Event) (bool, error) {
+			switch event.Type {
+			case watch.Added, watch.Modified:
+				cv, ok := event.Object.(*configv1.ClusterVersion)
+				if !ok {
+					logrus.Warnf("Expected a ClusterVersion object but got a %q object instead", event.Object.GetObjectKind().GroupVersionKind())
+					return false, nil
+				}
+				if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, configv1.OperatorAvailable) {
+					logrus.Debug("Cluster is initialized")
+					return true, nil
+				}
+				if cov1helpers.IsStatusConditionTrue(cv.Status.Conditions, configv1.OperatorFailing) {
```

Comment:
Should we expand this to accept `OperatorProgressing` too?  Or does that not have useful messages?

Reply:
There does not appear to have anything more useful in the `Progressing` condition. These are the messages that I have seen in that condition.

When `Failing` is `False`,
```
Working towards 4.0.0-0.alpha-2019-01-30-125030
```

When `Failing` is `True`,
```
Unable to apply 4.0.0-0.alpha-2019-01-30-125030: the update could not be applied
```